src/python/grpcio_tests/tests/unit/_metadata_test.py

```diff
@@ -237,6 +238,81 @@ def testStreamStream(self):
                                              call.trailing_metadata()))
 
 
+class MetadataTypeTest(unittest.TestCase):
+    """Tests for the metadata type"""
+
+    _DEFAULT_DATA = (("key1", "value1"), ("key2", "value2"))
+    _MULTI_ENTRY_DATA = (("key1", "value1"), ("key1", "other value 1"),
+                         ("key2", "value2"))
+
+    def test_init_metadata(self):
+        test_cases = {
+            "emtpy": (),
+            "with-data": self._DEFAULT_DATA,
```

Comment:
IMO we should add here the `_MULTI_ENTRY_DATA` option, and most likely the test will fail if I'm not wrong.

Reply:
You're right, I've added your suggestion as another test scenario.